src/service/committers.svc.ts

```diff
@@ -0,0 +1,239 @@
+ * @returns Formatted overall statistics section
+ */
+export function formatOverallStats(overallStats: AuthorCommitCounts, grandTotal: number): string {
+  let report = '\n## Overall Statistics\n';
+
+  const sortedStats = Object.entries(overallStats).sort((a, b) => b[1] - a[1]);
+
+  for (const [author, count] of sortedStats) {
+    report += `${count.toString().padStart(6)}  ${author}\n`;
+  }
+
+  report += `${grandTotal.toString().padStart(6)}  GRAND TOTAL\n`;
+
+  return report;
+}
+
+/**
+ * Formats the report data as CSV
+ * @param data - The structured report data
+ */
+export function formatAsCsv(data: ReportData): string {
+  // First prepare all author names (for columns)
+  const allAuthors = new Set<string>();
+
+  // Collect all unique author names
+  for (const monthData of Object.values(data.monthly)) {
+    for (const author of Object.keys(monthData)) {
+      if (author !== 'total') allAuthors.add(author);
+    }
+  }
+
+  const authors = Array.from(allAuthors).sort();
+
+  // Create CSV header
+  let csv = `Month,${authors.join(',')},Total\n`;
+
+  // Add monthly data rows
+  const sortedMonths = Object.keys(data.monthly).sort();
+  for (const month of sortedMonths) {
+    csv += month;
+
+    // Add data for each author
+    for (const author of authors) {
+      const count = data.monthly[month][author] || 0;
+      csv += `,${count}`;
+    }
+
+    // Add monthly total
+    csv += `,${`${data.monthly[month].total}\n`}`;
+  }
+
+  // Add overall totals row
+  csv += 'Overall';
+  for (const author of authors) {
+    const count = data.overall[author] || 0;
+    csv += `,${count}`;
+  }
+
+  csv += `,${data.overall.total}\n`;
+
+  return csv;
+}
+
+/**
+ * Formats the report data as text
+ * @param data - The structured report data
+ */
+export function formatAsText(data: ReportData): string {
+  let report = 'Monthly Commit Report\n';
+
+  // Monthly sections
+  const sortedMonths = Object.keys(data.monthly).sort();
+  for (const month of sortedMonths) {
+    report += `\n## ${month}\n`;
+
+    const authors = Object.entries(data.monthly[month])
+      .filter(([author]) => author !== 'total')
+      .sort((a, b) => b[1] - a[1]);
+
+    for (const [author, count] of authors) {
+      report += `${count.toString().padStart(6)}  ${author}\n`;
+    }
+
+    report += `${data.monthly[month].total.toString().padStart(6)}  TOTAL\n`;
+  }
+
+  // Overall statistics
+  report += '\n## Overall Statistics\n';
+  const sortedEntries = Object.entries(data.overall)
+    .filter(([author]) => author !== 'total')
+    .sort((a, b) => b[1] - a[1]);
+
+  for (const [author, count] of sortedEntries) {
+    report += `${count.toString().padStart(6)}  ${author}\n`;
+  }
+
+  report += `${data.overall.total.toString().padStart(6)}  GRAND TOTAL\n`;
+
+  return report;
+}
+
+/**
+ * Formats the report data as JSON
+ * @param data - The structured report data
+ */
+export function formatAsJson(data: ReportData): string {
```

Comment:
```suggestion
export function formatAsJson(data: ReportData): string {
```

This is a passthrough we could avoid it

Reply:
Done 0bfff29